stdlib/public/core/Availability.swift

```diff
@@ -42,3 +42,20 @@ public func _stdlib_isOSVersionAtLeast(
   return false._value
 #endif
 }
+
+#if os(macOS)
+// This is a magic entry point known to the compiler. It is called in
+// generated code for API availability checking.
+@_semantics("availability.osversion")
+@_effects(readnone)
+public func _stdlib_isOSVersionAtLeastOrVariantVersionAtLeast(
```

Comment:
Could you provide some explanation for re-introducing this entry point?

Reply:
This is important for distinguishing which version number to use (macOS or iOS) for Apple Silicon Macs.

Reply:
Expanding on this: this is an entry point that macOS binaries assume is present in the Standard Library. Having it here ensures that this version of the Standard Library remains compatible with such binaries -- including the overlays that are now part of their corresponding framework.

Reply:
> Expanding on this: this is an entry point that macOS binaries assume is present in the Standard Library.

In other words, removing it in the past was a mistake because it was already committed to the stable ABI? But why only macOS and not other platforms?